include/xgboost/data/simple_csr_source.h

```diff
@@ -47,6 +47,9 @@ class SimpleCSRSource : public DataSource {
    * \param info The additional information reflected in the parser.
    */
   void CopyFrom(dmlc::Parser<uint32_t>* src);
+
+  // Additional method to support multiple files
+  void CopyFromMultiple(std::vector<std::shared_ptr<dmlc::Parser<uint32_t>>> parsers, int num_parsers);
```

Comment:
Should we overload the `CopyFrom` function instead?

Reply:
Done